pkg/endpoint/policy.go

```diff
@@ -978,9 +978,18 @@ func (e *Endpoint) runIPIdentitySync(endpointIP netip.Addr) {
 				}
 
 				ID := e.SecurityIdentity.ID
-				hostIP, ok := netipx.FromStdIP(node.GetIPv4())
-				if !ok {
-					return controller.NewExitReason("Failed to convert node IPv4 address")
+				var hostIP netip.Addr
+				var ok bool
+				if endpointIP.Is4() {
+					hostIP, ok = netipx.FromStdIP(node.GetIPv4())
+					if !ok {
+						return controller.NewExitReason("Failed to convert node IPv4 address")
+					}
+				} else {
+					hostIP, ok = netipx.FromStdIP(node.GetIPv6())
+					if !ok {
+						return controller.NewExitReason("Failed to convert node IPv6 address")
+					}
```

Comment:
Please split this change into a separate patch, and consider how to test it.

I'm not familiar with the details of this codepath, but I'm slightly surprised about the logic you're proposing. Why does the address family of the endpoint automatically imply the node's address family? Can you describe how this will behave in a dual-stack environment?